docs/developer-guide/ksqldb-reference/scalar-functions.md

```diff
@@ -734,6 +734,86 @@ multiple elements, like those containing wildcards, aren't supported.
 
     `CREATE STREAM LOGS (LOG STRUCT<CLOUD STRING, APP STRING, INSTANCE INT>, ...) WITH (VALUE_FORMAT='JSON', ...)`
 
+### `IS_JSON_STRING`
+
+Since: 0.25.0
+
+```sql
+is_json_string(json_string) -> Boolean
+```
+
+Given a string, returns `true` if it can be parsed as a valid JSON value, `false` otherwise.
+
+Examples:
+
+```sql
+is_json_string("[1, 2, 3]") => true
+is_json_string("{}") => true
+is_json_string("1") => true
+is_json_string("\"abc\"") => true
+is_json_string("null") => true
+is_json_string("") => false
+is_json_string("abc") => false
+is_json_string(NULL) => false
+```
+
+### `JSON_RECORDS`
+
+Since: 0.25.0
+
+```sql
+json_records(json_string) -> Array<Struct<json_key:String, json_value:String>>
```

Comment:
This may have already be covered earlier, but could this return a Map<Key,Key> instead?

Reply:
The discussion about return types is in the KLIP - https://github.com/confluentinc/ksql/pull/8550#discussion_r779937714. If you prefer, we can discuss it there with a wider audience.

Reply:
Nice; after I started reviewing, I realized that there was a KLIP for this.  (And that I'm late to the party; thanks for the link!)

I'm not sure where the discussion should be between the two places.

Reply:
You are definitely not too late! We can continue the discussion in the KLIP or here. In my view, a `Struct` is a strictly better type than a `Map<String, String>` in this specific case because the former provides more comprehensive information and leaves less room for errors (and therefore error checking.) 

For example, consider a user program that receives `Map<String, String>` from a function. A careful programmer has to check the size of the map and that it contains promised key-value pairs. On the other hand, the struct type is self-documenting - there will be only 2 keys of the given type. WDYT?

Reply:
I'm not sure.  I think that `Map<String,String>` and `Array<Struct<String,String>>` are equivalent types in some sense.  If I have a map, I ought to be able to iterate over it like an Array, and if given an Array, one will invariably want to do something that looks like a `map.get()`.

Reply:
I just realized that your argument is about using `Map<String,String>` as the return type rather than a replacement for the `Struct` in the array, i.e. `Array<Map<String, String>>`. Sorry about that! 

I agree that `Map<String, String>` has as much type information as `Array<Struct<String,String>>`, so it could be used instead. I really don't have any strong arguments one way or another. As a kinda weak argument, its neighbour function, `json_keys`, returns an array, so it is probably more consistent to return an array here as well assuming everything else is equal.

I don't have a strong opinion here, so if you think that `Map<String, String` is a better return type, let's discuss it with a wider audience in [this thread](https://github.com/confluentinc/ksql/pull/8550#discussion_r781239602) in the KLIP.